datafusion/physical-optimizer/src/enforce_sorting/mod.rs

```diff
@@ -119,7 +119,7 @@ fn update_sort_ctx_children(
     }
 
     node.data = data;
-    node.update_plan_from_children()
```

Comment:
You also could have moved this to the start of this method, and change it's meaning to also sync the children. But that doesn't really matter. 🤷🏼